### PR TITLE
Resolve #521 - Fix cell array overflow

### LINKED ISSUE #521: NavGrid:GetCell - Ranged abilities can exceed length of Cells array at certain locations, causing crashes
Using ranged abilities (including flash, Ezreal E, etc) aimed past the northern wall on the 4.20 SR causes crashes.
screen05

Cells.Length: 86142
Attempted to grab at index: 86382

GetCell is called and attempts to fetch Cells[86185] (and often higher) when the array only has 86142 elements, thus causing an index out of bounds exception. We can simply ignore this and log to console when it happens (which is how I was debugging it) but that could have unintended side effects...

I have not tested this on other maps.

### PR DESCRIPTION
**Pull request by FurkanS1821**
Ranged abilities will no longer exceed the length of the Cell array and make the server crash.
Also updated gitignore because my git wasn't ignoring the folder called ".vs"

Refs #521

### CODE REVIEW

File: GameServerLib/Logic/Content/NavGrid.cs [hunk 1]
```diff
@@ -226,12 +226,13 @@ public int GetCellIndex(float x, float z)
 
         public NavGridCell GetCell(short x, short y)
         {
-            if (x < 0 || x > XCellCount || y < 0 || y > YCellCount)
+            var index = y * XCellCount + x;
+            if (x < 0 || x > XCellCount || y < 0 || y > YCellCount || index > Cells.Length)
```
Comment (molenzwiebel) [inline GameServerLib/Logic/Content/NavGrid.cs]: This needs to be `index >= Cells.Length`

**Review by molenzwiebel — DISMISSED** [review]

**Comment by indiemini** [comment]
This works, but it may have unintended side effects for some abilities, especially ground targeted abilities. ie Using Lux E or Trundle pillar could behave unexpectedly.

**Comment by FurkanS1821** [comment]
Well mentioned, I will think about what can be done.

**Comment by FurkanS1821** [comment]
After a small research, I saw that it won't be a problem.

- Lux E cannot be cast outside of the map, neither Trundle E or any other spell.
- Any spell going off map won't be a problem with this fix, as collision handler will ignore them.
- `GetClosestTerrainExit()` will work even if spell is casted out of the map somehow, as `IsWalkable()` will work properly even if cell is null.

**Review by MythicManiac — APPROVED** [review]
I no longer see any issues with the changes :shipit: 

The title is using a single dash to separate issue/title when we've used double dash in the past, so I'll squash merge this one with that changed

Good work :+1:

**Pull request merged**